src/common.cpp

```diff
@@ -1898,6 +1901,63 @@ std::string get_executable_path(const char *argv0) {
         }
         return buffstr;
     }
+#else
+    static std::string result;
+    if (!result.empty()) return result;
+    std::string strargv0 = ((argv0) ? argv0 : ""), buffstr;
+    struct stat st;
+    if (!strargv0.empty()) {
+        if (strargv0[0] == '/') {
+            buffstr = strargv0;
+        } else if (buffstr.find('/') == std::string::npos) {
+            const char *penv = getenv("PATH");
+            if (penv && *penv) {
+                std::vector<std::string> env = split_string(penv, ':');
+                for (std::size_t i = 0; i < env.size(); i++) {
+                    buffstr = env[i] + "/" + strargv0;
+                    if (!stat(buffstr.c_str(), &st) && (st.st_mode & S_IXUSR) && (st.st_mode & S_IFREG)) {
+                        break;
+                    }
+                }
+            }
+        } else {
+            const char *pwd = getenv("PWD");
+            if (pwd && *pwd) {
+                buffstr = std::string(pwd) + "/" + strargv0;
+            }
+        }
+        return buffstr;
+    }
+#ifdef __OpenBSD__
+    // kd variable is necessary for kvm_getfiles() first argument; RTFM.
+    kvm_t *kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, nullptr);
+    if (!kd) { result = "fish"; return result; }
+    int cntp = 0;
+    bool ok = false;
+    kinfo_file *kif = nullptr;
+    if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, getpid(), sizeof(struct kinfo_file), &cntp))) {
+        if (!stat(buffstr.c_str(), &st)) {
```

Comment:
Can this actually be *reached*?

I don't see any way `buffstr` gets any value without being returned?